### ISSUE TITLE
Crashes on Loading Scenario without Error Message

### ISSUE THREAD

**Issue by RogerCooper** [triplea-game/triplea#2703]
### My Operating System:
Windows X
### Engine version:
1.9.0.0.7621
### Map name and version:
3rd Century BCE Wars
### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
Click the play button
### Do you have the exact error text? Please copy/paste if so
No text. Goes back to load screen.
### Instead of this error, what should have happened?
It should show the error log.
### Any additional information that may help:
[AnicentBCffa.zip](https://github.com/triplea-game/triplea/files/1566108/AnicentBCffa.zip)

**Comment by ssoloff** [comment]
Here's the exception that's being raised:

```
java.lang.reflect.InvocationTargetException
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1321)
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1296)
	at javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1348)
	at games.strategy.triplea.TripleA.startGame(TripleA.java:122)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.launchInNewThread(LocalLauncher.java:49)
	at games.strategy.engine.framework.startup.launcher.AbstractLauncher.lambda$0(AbstractLauncher.java:51)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.UnsupportedOperationException
	at java.util.AbstractList.remove(AbstractList.java:161)
	at games.strategy.triplea.ui.mapdata.MapData.getPlayerColor(MapData.java:444)
	at games.strategy.triplea.ui.screen.drawable.LandTerritoryDrawable.draw(LandTerritoryDrawable.java:29)
	at games.strategy.triplea.ui.screen.SmallMapImageManager.updateTerritoryOwner(SmallMapImageManager.java:79)
	at games.strategy.triplea.ui.MapPanel.initSmallMap(MapPanel.java:740)
	at games.strategy.triplea.ui.TripleAFrame.<init>(TripleAFrame.java:258)
	at games.strategy.triplea.TripleA.lambda$0(TripleA.java:124)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:353)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

The property `color.Scythians` is not defined, and the code attempts to pull off the next available default color.  However, because the default color list is immutable, this operation fails, and the exception is raised.

The fact that the default color list is immutable might be a regression.  I'll look into it further.

**Comment by ssoloff** [comment]
Yeah, looks like support for default colors broke in 6d7783c.  I'll submit a fix shortly.

I'm not sure if there's a workaround without the code fix because I don't really understand how map variants work.  But if there's a way to define the map property `color.Scythians` in the 3rdCentury_BCE_Wars map (and any other new player color not defined in 270BC), that should prevent the exception from being raised.

**Comment by ssoloff** [comment]
Note that after fixing the default colors issue, the following exception is now raised:

```
java.lang.reflect.InvocationTargetException
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1321)
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1296)
	at javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1348)
	at games.strategy.triplea.TripleA.startGame(TripleA.java:122)
	at games.strategy.engine.framework.startup.launcher.LocalLauncher.launchInNewThread(LocalLauncher.java:49)
	at games.strategy.engine.framework.startup.launcher.AbstractLauncher.lambda$0(AbstractLauncher.java:51)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Image Not Found:flags/GermanicTribes_small.png
	at games.strategy.triplea.image.ImageFactory.getImage(ImageFactory.java:35)
	at games.strategy.triplea.image.ImageFactory.getImage(ImageFactory.java:28)
	at games.strategy.triplea.image.FlagIconImageFactory.getSmallFlag(FlagIconImageFactory.java:26)
	at games.strategy.triplea.ui.CommentPanel.createComponents(CommentPanel.java:98)
	at games.strategy.triplea.ui.CommentPanel.init(CommentPanel.java:62)
	at games.strategy.triplea.ui.CommentPanel.<init>(CommentPanel.java:58)
	at games.strategy.triplea.ui.TripleAFrame.<init>(TripleAFrame.java:261)
	at games.strategy.triplea.TripleA.lambda$0(TripleA.java:124)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:353)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

I don't know if there's supposed to be support for providing a default small flag if the map doesn't provide one.  Otherwise, the 3rdCentury_BCE_Wars map will need to include one for the new GermanicTribes player (and presumably one for any other new player not present in 270BC).

**Comment by RogerCooper** [comment]
The problem here is that the error message is not visible. In earlier versions of TripleA, I could always view the error log.

**Comment by ssoloff** [comment]
@RogerCooper Sorry for the misunderstanding.  I believe you and others have pointed this out before, so maybe we just need to bite the bullet and re-introduce the **View Error Console** button on the main menu.  (I believe it was removed for usability reasons, but maybe that reasoning didn't consider map makers.)

In the meantime, as a workaround, you can run TripleA from the command line with the `-console` switch:

```
C:\Users\me\Downloads\TripleA_1.9.0.0.7621>TripleA.exe -console
```

That will echo the TripleA Console output to the terminal so you can see the errors your map is generating.

**Comment by ron-murhammer** [comment]
@ssoloff Yeah, I believe the color error was a regression but the missing image is a valid error.

**Comment by DanVanAtta** [comment]
No disagreements on regression, but I'd suggest it was a bug to begin with to have ` final List<Color>`, the `final` makes it seem like we have an immutable data object, when that is not the case. Can we see if we can find the place that changes the list elements and avoid that from happening when fixing this? We can probably find it pretty easily by making the List a true immutable. Alternatively, let's remove the 'final' so that the code does not pretend we are dealing with immutables.

Second, is Arrays.asList() immutable? I didn't think so - I'm not sure how we are actually changing things by removing an unnecessary array initialization.

**Comment by RoiEXLab** [comment]
@DanVanAtta Arrays.asList actually returns a fixed size list AFAIK, so the content can be replaced but not be removed.
This issue was fixed in #2704 we slightly changed the under-the-hood behaviour, so we only use immutable data types now.

**Comment by ssoloff** [comment]
@RogerCooper As of build 1.9.0.0.7874, there is a new setting you can enable that will display the console when _any_ message is written to it, not just ones deemed errors:

![show-console-always-setting](https://user-images.githubusercontent.com/4826349/34181565-a9a631c0-e4e0-11e7-97b2-30710d872229.png)

Enabling this setting will cause the console to be displayed in the scenario you originally reported above.
